src/Platform/Microsoft.Testing.Platform/CommandLine/ResponseFileHelper.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+using Microsoft.Testing.Platform.Resources;
+
+// Most of the core logic is from:
+// - https://github.com/dotnet/command-line-api/blob/feb61c7f328a2401d74f4317b39d02126cfdfe24/src/System.CommandLine/Parsing/CliParser.cs#L49
+// - https://github.com/dotnet/command-line-api/blob/feb61c7f328a2401d74f4317b39d02126cfdfe24/src/System.CommandLine/Parsing/StringExtensions.cs#L349
+internal static class ResponseFileHelper
+{
+    internal static bool TryReadResponseFile(string rspFilePath, List<string> errors, [NotNullWhen(true)] out string[]? newArguments)
+    {
+        try
+        {
+            newArguments = ExpandResponseFile(rspFilePath).ToArray();
+            return true;
+        }
+        catch (FileNotFoundException)
+        {
+            errors.Add(string.Format(CultureInfo.InvariantCulture, PlatformResources.CommandLineParserResponseFileNotFound, rspFilePath));
+        }
+        catch (IOException e)
+        {
+            errors.Add(string.Format(CultureInfo.InvariantCulture, PlatformResources.CommandLineParserFailedToReadResponseFile, rspFilePath, e.Message));
```

Comment:
It might be worth adding the tostring of the exception.

Reply:
We could also as follow-up if we don't want to change all the callers/tests right now.